### PR TITLE
Validate data

### PR DESCRIPTION
**Pull request by xion6136**
Validation and back button should be working properly. Start/End time still need some work though.

### CODE REVIEW

**Review by ZackSchuster600 — APPROVED** [review]

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 1]
```diff
@@ -225,31 +274,219 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
...
+        // Validation outcome for class name
+        boolean testClassName = true;
+        testClassName = validateClassName();
+
+        // Validation outcome for the start and end time
+        boolean testTime = true;
+        testTime = validateTime();
+
+        // Check all final validation outcome, if any result to false, then set final result to false
+        if(testDaysOfTheWeek == false || testDepartmentName == false || testClassNumber == false || testClassName == false || validateTime() == false) {
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: Don't necessarily need to do it like this. You could just do

return (testDaysOfTheWeek == false || testDepartmentName == false || testClassNumber == false || testClassName == false || validateTime() == false) ;

If you wanted it even shorter you could just do:

return validateDates() && validateDepartment() && validateClassNumber() && validateClassName() && validateTime()

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 2]
```diff
@@ -225,31 +274,219 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
...
+
+    /**
+     * This function validate the department of the newly added class. If nothing is selected,
+     * then it will prompt the user to select a department.
+     * @return false is nothing is selected, else return true
+     */
+    private boolean validateDepartment() {
+        boolean result = true;
+        // If the class name has not been selected
+        if(dept_name.getSelectionModel().isEmpty()) {
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: add a space between the if and the (
Required by the style guide. This should be done for every if you have in this PR

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 3]
```diff
@@ -225,31 +274,219 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
...
+    /**
+     * This function validate that at least 1 of the day of the week is selected. If not
+     * then it will prompt the user to click on at least 1 or more day/days of the week.
+     * @return false is nothing is selected, else return true
+     */
+    private boolean validateDates() {
+        boolean result = false;
+        // Iterate through all 5 days to check if at least 1 day is checked.
+        // If so, set the boolean to true and break.
+        for(RadioButton day : datesSelected) {
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: add a space between the for and the (
Similar to the if

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 4]
```diff
@@ -225,31 +274,219 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
...
+        }
+
+        if(result == false) {
+            invalidStartAndEndTime.showAndWait();
+        }
+
+        return result;
+    }
+
+    // 1:00, 12:00
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: If this code is not needed, then it should be deleted and not included in the PR

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 5]
```diff
@@ -146,6 +171,29 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             e.printStackTrace();
         }
 
+        // Initialize the invalid error and hide them.
+        invalidClassName.setTitle("Invalid Class Name Error");
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: Do we need to initialize and then hide them all here? Or can we initialize them when we actually create the error?

**Review by dtbielejeski — CHANGES_REQUESTED** [review]

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 6]
```diff
@@ -305,36 +277,7 @@ public LocalDateTime convertToLocalDateTimeViaInstant(Date dateToConvert) {
...
-        testTime = validateTime();
-
-        // Check all final validation outcome, if any result to false, then set final result to false
-        if(testDaysOfTheWeek == false || testDepartmentName == false || testClassNumber == false || testClassName == false || validateTime() == false) {
-            result = false;
-        }
-
-        // Return result
-        return result;
+        return validateDates() && validateDepartment() && validateClassNumber() && validateClassName();
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: I think validateTime() is missing

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 7]
```diff
@@ -394,124 +346,25 @@ private boolean validateDates() {
...
 
         return result;
     }
 
-    /**
-     * This function validate that the user has inputed the start and end time in the correct 24-hour time format
-     * and that end time must be after start time. Otherwise, it will be invalid.
-     * @return false is time is incorrect, else return true
-     */
-    private boolean validateTime() {
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: Do we still need time validation?

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java (hunk 1 above)
Comment (xion6136) [reply ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: Sounds good! Thank you!

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java (hunk 2 above)
Comment (xion6136) [reply ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: Got it! Thanks!

File: ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java [hunk 8]
```diff
@@ -365,6 +370,79 @@ private boolean validateDates() {
         return result;
     }
 
+    private boolean validateTime() {
+        // TODO: time error showing up twice. Also need to validate that end time must be after start time and that it is an integer that is being entered
+        boolean result = true;
+        String startTime = start_time.getText();
+        int startTimeValue = -1;
+
+        if(startTime.length() == 0) {
```
Comment (dtbielejeski) [inline ClassySchedule/src/main/java/scheduler/AddCourseToScheduleController.java]: add spaces after if

**Review by dtbielejeski — APPROVED** [review]

**Pull request merged**